internal/store/store.go

```diff
@@ -35,12 +34,15 @@ func New(reader io.Reader, logger logrus.FieldLogger) (*Store, error) {
 
 func validatePlugins(plugins []*model.Plugin) error {
 	for _, plugin := range plugins {
-		if plugin.Manifest.Id == "" {
-			return errors.Errorf("plugin manifest Id is empty %+v", plugin)
+		err := plugin.Manifest.IsValid()
```

Comment:
Maybe the manifest should be even checked before the plugin gets added e.g. in `cmd/generator/main`. Thoughts?

Reply:
Agree, I think we should do it in both places. The reason for being both is that when parsing in the plugins.json file we can't guarantee that the data will be valid. And we should do it when we generate because we don't want to write invalid manifest to plugins.json